### PR TITLE
v1.12.0 – Updating more docs – Dev Principles, Getting Started and Gulp pages updated

### PR DESCRIPTION
**Pull request by ashleynolan**
### Added
- CODEOWNERS file added
- PR Template added

### Updated
- Development Principles docs page up-to-date
- Getting Started docs page up-to-date
- Gulp docs page up-to-date

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,12 +20,12 @@
   },
```
Comment (xander-marjoram) [inline package.json]: Version bump please

File: docs/src/templates/pages/documentation/general/getting-started.hbs [hunk 2]
```diff
@@ -1,35 +1,33 @@
...
 navsub: docs-general
 navactive: getting-started
-
-wip: true
 ---
 
-We have tried to decouple our JUST EAT’s front-end process from our .NET stack so that developing front-end code can be fast and iterative.  This also enables our front-end developers to work in the environment that is most productive for them.
 
-Our current front-end stack uses Gulp (as our task runner), Sass and PostCSS (for our CSS processing) and Browserify with Babel (for our JavaScript).
+We have tried to decouple our UI stack away from out back-end (.NET) stack, so that developing front-end code can be fast and iterative. This also enables our front-end developers to work in the environment that is most productive for them.
```
Comment (xander-marjoram) [inline docs/src/templates/pages/documentation/general/getting-started.hbs]: out back-end -> our back-end

File: docs/src/templates/pages/documentation/general/getting-started.hbs [hunk 3]
```diff
@@ -1,35 +1,33 @@
...
+
+If you are going to be developing front-end code for the Just Eat platform, it’s recommended that you take the time to read through this documentation to get a better understanding of the tools we use and why we are using them.
 
-If you are going to be developing front-end code for the JUST EAT platform, it’s recommended that you take the time to read all the sections of this documentation to get a better understanding of the tools we use and why we are using them.
 
+## Web Stack
 
-<a name="proj-setup"></a>
-## Setting up your Environment
+Fozzie itself has been created to be fairly technology agnostic.  At it's core, it's simply a UI framework built on Sass and Vanilla JS.
```
Comment (xander-marjoram) [inline docs/src/templates/pages/documentation/general/getting-started.hbs]: it's core -> its core

File: docs/src/templates/pages/documentation/general/getting-started.hbs [hunk 4]
```diff
@@ -1,35 +1,33 @@
...
+We also use Yarn for our package dependencies.
 
-<a name="proj-gulp"></a>
-## Front-end Workflow – Gulp
 
-Gulp is a front-end task runner that automates many important tasks associated with modern front-end development.
+## Front-end Build Tasks
 
-We use Gulp on the JUST EAT platforms to tie together all of our front-end workflow tasks.  To find out more about Gulp and how we use it, take a lookat the [Gulp section of this documentation]({{ baseUrl }}/documentation/general/gulp).
+Gulp is a front-end task runner that automates tasks such as Sass compilcation and Babel transpilation.
```
Comment (xander-marjoram) [inline docs/src/templates/pages/documentation/general/getting-started.hbs]: compilcation -> compilation

File: docs/src/templates/pages/documentation/general/gulp.hbs [hunk 5]
```diff
@@ -1,169 +1,120 @@
...
-## Using Gulp in Visual Studio
+For details on how to use and figure this module, take a look at the [module's documentation](https://www.npmjs.com/package/@justeat/gulp-build-fozzie#setup).
 
-Here we document the steps needed to successfully run Gulp using Visual Studio 2013 and 2015.
 
-It is recommended that you upgrade to VS 2015 as the extension required to run Gulp tasks – Task Runner Explorer – comes baked in by default.
+<a name="gulp-setup"></a><h2 class="sg-sectionHeading">Getting up and running with Gulp</h2>
 
-### Setup steps
+Here we document two ways of running Gulp tasks locally: using Visual Studio and it’s built-in tools, or running Gulp through the command line.
```
Comment (xander-marjoram) [inline docs/src/templates/pages/documentation/general/gulp.hbs]: it's -> its

File: docs/src/templates/pages/documentation/general/gulp.hbs [hunk 6]
```diff
@@ -1,169 +1,120 @@
...
 
 1. **[Install Node.js](https://nodejs.org/)**
 
-	Node.js – and it’s bundled package manage npm – are needed to run Gulp tasks with VS 2013 or VS 2015 Professional.
+    Node.js – and it’s bundled package manager npm – are needed to run Gulp tasks with VS 2013 or VS 2015 Professional.
 
-	Installing this should be as simple as going to the [Node.js website](https://nodejs.org/) and hitting the 'Install' button, which will take you through the install process.  Go with all the defaults throughout the install process and this should install Node and it’s package manager on your system.
-
-2. **For users of VS 2013 only** – **[Install Task Runner Explorer](https://visualstudiogallery.msdn.microsoft.com/8e1b4368-4afb-467a-bc13-9650572db708)**
+    Installing this should be as simple as going to the [Node.js website](https://nodejs.org/) and hitting the 'Install' button, which will take you through the install process.  Go with all the defaults throughout the install process and this should install Node and it’s package manager on your system.
```
Comment (xander-marjoram) [inline docs/src/templates/pages/documentation/general/gulp.hbs]: it's package manager -> its package manager

File: docs/src/templates/pages/documentation/general/gulp.hbs [hunk 7]
```diff
@@ -1,169 +1,120 @@
...
 
-	We recommend that you upgrade to Visual Studio 2015 rather than doing this, but if you really want to keep using VS 2013 then you’ll need to install the [Task Runner Explorer extenson](https://visualstudiogallery.msdn.microsoft.com/8e1b4368-4afb-467a-bc13-9650572db708).
+2. **Install Gulp**
 
-3. **Install Gulp**
+    Open up a command prompt or terminal window and enter `npm install gulp -g`.  This should install Gulp so that Visual Studio can do it’s magic.
```
Comment (xander-marjoram) [inline docs/src/templates/pages/documentation/general/gulp.hbs]: it's magic -> its magic

File: docs/src/templates/pages/documentation/general/gulp.hbs [hunk 8]
```diff
@@ -1,169 +1,120 @@
...
 
-<hr class="sectionSplitter">
+To get up and running with using Gulp in your setup of Visual Studio, just carry out the following steps:
 
-<a name="gulp-tasks"></a>
-## Gulp Tasks
+1. **[Install Node.js](https://nodejs.org/)**
 
-All tasks can be run from Visual Studio by right clicking the name of the task, and selecting **Run**.
+	Node.js – and it’s bundled package manager npm – are needed to run Gulp tasks with VS 2013 or VS 2015 Professional.
```
Comment (xander-marjoram) [inline docs/src/templates/pages/documentation/general/gulp.hbs]: it's -> its

File: docs/src/templates/pages/documentation/general/gulp.hbs [hunk 9]
```diff
@@ -1,169 +1,120 @@
...
 
----
+	Installing this should be as simple as going to the [Node.js website](https://nodejs.org/) and hitting the 'Install' button, which will take you through the install process.  Go with all the defaults throughout the install process and this should install Node and it’s package manager on your system.
```
Comment (xander-marjoram) [inline docs/src/templates/pages/documentation/general/gulp.hbs]: it's -> its

File: docs/src/templates/pages/documentation/general/gulp.hbs [hunk 10]
```diff
@@ -1,169 +1,120 @@
...
 
-### gulp default
+2. **Install Gulp**
 
-The default Gulp task is made up of 3 other tasks.  It will run the [css](#gulp-css), [scripts](#gulp-scripts) and [copy:fonts](#gulp-copy) Gulp tasks, which will give you a compiled version of the front–end code for use in your local build.
+	Open up a command prompt or terminal window and enter `npm install gulp -g`.  This should install Gulp so that Visual Studio can do it’s magic.
```
Comment (xander-marjoram) [inline docs/src/templates/pages/documentation/general/gulp.hbs]: it's -> its

File: docs/src/templates/pages/documentation/general/gulp.hbs [hunk 11]
```diff
@@ -1,169 +1,120 @@
...
+	In the Visual Studio Menu Bar, go to `Tools –> Options –> Projects and Solutions –> External Web Tools`.  Then, move the `$(PATH)` reference in the displayed list to the top of the list.  It should [look like this](http://i.imgur.com/SmzYdZs.png).
 
-This task currently creates 4 CSS files; je.css, je-old.css, opinion-labs.css and styleguide.css.
+    This prioritises your own locally installed version of Node over that which comes along with Visual Studio (which can become out-of-date).
 
-For more details on the CSS structure, see the [CSS documenation]({{ baseUrl }}/css/) which goes into more detail.
+…and that’s it!
 
----
+If you don’t plan on editing any CSS yourself, you can forget Gulp even exists and it’ll automatically do it’s thing when you build the project through Visual Studio.
```
Comment (xander-marjoram) [inline docs/src/templates/pages/documentation/general/gulp.hbs]: it's -> its

File: docs/src/templates/pages/documentation/general/gulp.hbs [hunk 12]
```diff
@@ -1,169 +1,120 @@
...
-We will be doing more with our fonts, in terms of caching and optimisation, in the future and so this task is here as a means to enable this process later.
 
----
+<a name="gulp-tasks"></a><h2 class="sg-sectionHeading">Available Tasks</h2>
 
-### gulp watch
+To see a full list of the tasks that are available as part of the `gulp-build-fozzie` module, [check out the module documentation](https://github.com/justeat/gulp-build-fozzie#the-gulp-tasks).
 
-The `gulp watch` task allows you to set a watcher on your files, so that when a SCSS or JavaScript file changes, the relevant Gulp task with run automatically.
+Note that some tasks will require some configuration – such as telling the module which directories should be compiled.  All of these configuartion details are explained in the documentation as well.
```
Comment (xander-marjoram) [inline docs/src/templates/pages/documentation/general/gulp.hbs]: configuartion -> configuration

File: package.json (hunk 1 above)
Comment (ashleynolan) [reply package.json]: Done ⬆️ 🙃

File: docs/src/templates/pages/documentation/general/gulp.hbs (hunk 10 above)
Comment (ashleynolan) [reply docs/src/templates/pages/documentation/general/gulp.hbs]: I clearly need training on the use of its vs it’s 😉

**Comment by ashleynolan** [comment]
@xander-marjoram Review comments looked at – let me know if you find anything else 😄

File: docs/src/templates/pages/documentation/general/gulp.hbs [hunk 13]
```diff
@@ -1,169 +1,120 @@
...
 
 1. **[Install Node.js](https://nodejs.org/)**
 
-	Node.js – and it’s bundled package manage npm – are needed to run Gulp tasks with VS 2013 or VS 2015 Professional.
-
-	Installing this should be as simple as going to the [Node.js website](https://nodejs.org/) and hitting the 'Install' button, which will take you through the install process.  Go with all the defaults throughout the install process and this should install Node and it’s package manager on your system.
+    Node.js – and its bundled package manager npm – are needed to run Gulp tasks with VS 2013 or VS 2015 Professional.
 
-2. **For users of VS 2013 only** – **[Install Task Runner Explorer](https://visualstudiogallery.msdn.microsoft.com/8e1b4368-4afb-467a-bc13-9650572db708)**
+    Installing this should be as simple as going to the [Node.js website](https://nodejs.org/) and hitting the 'Install' button, which will take you through the install process.  Go with all the defaults throughout the install process and this should install Node and its package manager on your system.
```
Comment (DamianMullins) [inline docs/src/templates/pages/documentation/general/gulp.hbs]: There isn't an install button on the Node website, maybe expand this sentence a little by specifying that they should download and install the latest LTS version indicated on the homepage?

File: docs/src/templates/pages/documentation/general/gulp.hbs [hunk 14]
```diff
@@ -1,169 +1,120 @@
...
-
-<a name="gulp-cmd"></a>
-## Using Gulp on Command Line
-
-Another method of using Gulp is straight through the Command Line or Terminal.  If you’ve already followed the installation steps above, you should be able to do this by running `gulp` from the `/src/ConsumerWeb/` directory.
+You should now be able to run `gulp` from the directory containing the projects `package.json`.
 
 By default, Gulp will run the default task, but you can run any of the project tasks by appending the name of the task to your `gulp` call. For example, running`gulp css` will run the task that compiles the CSS files.
 
+For a full list of tasks avilable, [take a look at the modules documentation](https://github.com/justeat/gulp-build-fozzie#the-gulp-tasks).
```
Comment (DamianMullins) [inline docs/src/templates/pages/documentation/general/gulp.hbs]: Alternatively, they can type `gulp --tasks` which will output a list of all available tasks to the command line.

Comment (ashleynolan) [reply docs/src/templates/pages/documentation/general/gulp.hbs]: Have added in a line with this info in :smil

File: docs/src/templates/pages/documentation/general/gulp.hbs (hunk 13 above)
Comment (ashleynolan) [reply docs/src/templates/pages/documentation/general/gulp.hbs]: Good catch – thought I'd get away with just copying this bit… 🙄

**Pull request merged**